### ISSUE TITLE
updating parameters in a batch

### ISSUE THREAD

**Issue by miangoleh** [mitsuba-renderer/mitsuba3#1587]
I am trying to only perform the opt.step() after accumulating grads from a few different samples. 
```
for iteration
  for views
    compute loss
    loss backwards
   opt step
```

However by probing the opt it seems that gradients are not accumulated and only the first computation is kept. Is there a way to achieve the similar effect as of using batches/mini batches in training neural networks?
